docs/source/guide/ddd-5-theory.myst

```diff
@@ -4,12 +4,91 @@ jupytext:
+is a quantum control technique to effectively reduce the interaction of a quantum system with its environment.
+The protocol works by driving a quantum system with rapid sequences of periodic control pulses.
+The application of DD sequences can have in two effects depending on the correlation time of the environment:
+
+1. For Markovian noise, DD can make the overall quantum channel more symmetric (analogously to quantum twirling)
+but cannot actually decouple the system from the environment;
+
+2. For non-Markovian noise, DD can effectively decouple the system form the environment.
+In theory, for ideal sequences of infinitely quick and strong pulses, complete noise suppresion is achievable.
+
+In real practical settings, due to the finite frequency and finite amplitude of DD sequences,
+both effects are possible but only as imperfect approximations.
+
+In the context of quantum computing, DD can be considered as an error mitigation method.
+With respect to other error mitigation techniques, DDD has very peculiar features:
+
+- It maps a noisy quantum computation to a _single_ error-mitigated computation (no need to take linear combinations
+of noisy results as in [ZNE](zne-5-theory.myst), [PEC](pec-5-theory.myst) and [CDR](cdr-5-theory.myst)).
+
+- As a consequence of the previous point, there is not a fundamental error mitigation overhead or
+an increase of statistical uncertainty in the final results.
+
+- If noise is time-correlated, it can suppress real errors at the physical level instead of applying a virtual noise
+reduction via classical post-processing.
+
+
+
+
+## Digital dynamical decoupling
+
+In a quantum computing device based on the circuit model, sequences of DD pulses can be mapped to sequences
+of discrete quantum gates (typically Pauli gates). We refer to this gate-level formulation as _digital dynamical decoupling_ (DDD)
+to distinguish it from the standard pulse-level formulation.
+
+
+
+```{note}
+This type of gate-level approach is very similar to the gate-level abstraction used in Mitiq to implement
+_digital zero-noise extrapolation_ via _unitary folding_ (see [What is the theory behind ZNE?](zne-5-theory.myst)).
+```
+Experimental evidence showing the practical utility gate-level decoupling sequences is given in several publications {cite}`Pokharel_2018_PRL, Jurcevic_2021_arxiv, GoogleQuantum_2021_nature, Smith_2021_arxiv, Das_2021_ACM`.
+
+
+```{warning}
+Gate-level DDD can only be considered as an approximation of the ideal (pulse-level) DD technique. Moreover, quantum backends 
+may internally optimize and schedule gates in unpredictable ways such that, in practice, DDD sequences may not be physically applied
+as expected.
+```
+
+A significant advantage of DDD with respect to pulse-level DD is the possibility of applying it in a backend-independent way, 
+via simple transformations of abstract quantum circuits. For this reason, DDD is particularly suitable for a multi-platform library like Mitiq.
```

Comment:
This line seems to conflict with the warning given above. The warning seems to say that backends may alter your circuit and actually erase some of your "pulses" via compilation. Then these sentences say it's backend-independent?

Reply:
DDD can be **defined** in a back-end independent way since it acts on circuits and not on pulses. 
In practice however, the results of the computation can be backend dependent as you noticed.

So, in [9578789](https://github.com/unitaryfund/mitiq/pull/1311/commits/95787896e97791b4bc5ec369f2343db29ebdc55e), I changed the word _applying_ with _defining_ in that sentence.